org.eclipse.m2e.pde.ui/src/org/eclipse/m2e/pde/ui/target/editor/internal/operation/ModifyPropertyOperation.java

```diff
@@ -0,0 +1,59 @@
+/*******************************************************************************
+ * Copyright (c) 2023 Patrick Ziegler and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-v20.html
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *      Patrick Ziegler - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.m2e.pde.ui.target.editor.internal.operation;
+
+import java.util.function.Consumer;
+
+import org.eclipse.core.commands.ExecutionException;
+import org.eclipse.core.commands.operations.AbstractOperation;
+import org.eclipse.core.runtime.IAdaptable;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.IStatus;
+import org.eclipse.core.runtime.Status;
+
+/**
+ * This class updates the property of a single element within the model. This
+ * change operation <b>does not</> trigger side-effects. This behavior is
+ * required when e.g. editing a cell in the table viewer. The data model should
+ * be updated with the new value but the input and selection should not be
+ * updated explicitly, as this is already handled by JFace.
+ */
+public class ModifyPropertyOperation extends AbstractOperation {
```

Comment:
This class is also only used once and could be replaced by a corresponding anonymous class at the caller.

Reply:
If it helps to keep the code more focused I think a "extra" class is fine, but it could probably made package private to show it is only used here.